File: pkg/pwrgrep/testdata/fixtures/csharp/InputReachesEffect.cs
public class InputReachesEffect : ControllerBase
{
    void Run()
    {
        var cmd = Environment.GetEnvironmentVariable("CMD");
        var full = cmd + " --verbose";
        // ruleid: cs-input-reaches-effect
        Process.Start(full);
        // ok: cs-input-reaches-effect
        Process.Start("ls");
    }

    void Handle(SqlConnection conn)
    {
        var name = Request.Query["name"];
        // ruleid: cs-input-reaches-effect
        var bad = new SqlCommand("SELECT * FROM users WHERE name = '" + name + "'", conn);
        // ok: cs-input-reaches-effect
        var good = new SqlCommand("SELECT 1", conn);
    }
}

File: pkg/pwrgrep/testdata/fixtures/javascript/input-reaches-effect.ts
import { exec } from "child_process";

function run() {
  const cmd = process.env.CMD;
  const full = cmd + " --verbose";
  // ruleid: js-input-reaches-effect
  exec(full);
  // ok: js-input-reaches-effect
  exec("ls");
}

function handle(req, res, db) {
  const name = req.query.name;
  // ruleid: js-input-reaches-effect
  db.query("SELECT * FROM users WHERE name = '" + name + "'");
  // ok: js-input-reaches-effect
  db.query("SELECT 1");
}
